src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java

```diff
@@ -175,6 +179,19 @@ public void start() {
     asyncHttpClient = new AsyncHttpClient(new GrizzlyAsyncHttpProvider(config), config);
   }
 
+  private Scheduler getWorkerScheduler(SchedulerConfig config) {
+    if (streamingEnabled) {
+      return schedulerService.customScheduler(config.withMaxConcurrentTasks(getMaxStreamingWorkers()),
```

Comment:
investigate how many schedulers/trhreads may be created here on complex apps with lots of requester-configs. Create jira and add a TODO